### PR TITLE
[ JPG - 008] Delete Transaction

### PR DESCRIPTION
**Pull request by Aashishverma06**
Delete transaction

### CODE REVIEW

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 1]
```diff
@@ -19,13 +20,20 @@ public class TransactionService {
...
+     * @param userRepository        the user repository
+     */
     public TransactionService(TransactionRepository transactionRepository, UserRepository userRepository) {
         this.transactionRepository = transactionRepository;
         this.userRepository = userRepository;
     }
 
     /**
      * Create method to do the actual transaction contaning the business logic to add deposit and withdraw and return response according
+     *
```
Comment (Hrishikesh-SP18) [inline src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:36]: same here extra line

File: src/test/java/com/passbook/sparkeighteen/controller/TransactionControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.passbook.sparkeighteen.service.TransactionService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
```
Comment (Hrishikesh-SP18) [inline src/test/java/com/passbook/sparkeighteen/controller/TransactionControllerTest.java:12]: extra line

File: src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.passbook.sparkeighteen.peristence.POJO.TransactionType;
+import com.passbook.sparkeighteen.peristence.entity.TransactionEntity;
+import com.passbook.sparkeighteen.peristence.repository.TransactionRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
```
Comment (Hrishikesh-SP18) [inline src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java:11]: extra line

File: src/main/java/com/passbook/sparkeighteen/controller/TransactionController.java [hunk 4]
```diff
@@ -22,12 +23,18 @@
...
+     *
+     * @param transactionService the transaction service
+     */
     public TransactionController(TransactionService transactionService) {
         this.transactionService = transactionService;
     }
 
     /**
      * To return response entity of the transaction performed in transaction service
+     *
```
[the remark this answers is not in the document]
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/controller/TransactionController.java]: Removed.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 5]
```diff
@@ -19,13 +20,20 @@ public class TransactionService {
 
     private final UserRepository userRepository;
 
+    /**
+     * Instantiates a new Transaction service.
+     *
```
[the remark this answers is not in the document]
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java]: Removed.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java (hunk 1 above)
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java]: Removed.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 6]
```diff
@@ -53,4 +61,19 @@ private Float getUpdatedBalance(UserEntity user) {
         // TODO: Add getting the latest closing balance of the user
         return balance;
     }
+
+    /**
+     * Delete transaction string.
+     *
```
[the remark this answers is not in the document]
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java]: Removed.

File: src/test/java/com/passbook/sparkeighteen/controller/TransactionControllerTest.java (hunk 2 above)
Comment (Aashishverma06) [reply src/test/java/com/passbook/sparkeighteen/controller/TransactionControllerTest.java]: Removed.

File: src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java (hunk 3 above)
Comment (Aashishverma06) [reply src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java]: Removed.

**Comment by Ketan-SP18** [comment]
@Aashishverma06 is this a pr for update profile ?? please check your last commit message.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 7]
```diff
@@ -19,13 +20,18 @@ public class TransactionService {
...
+     * @param userRepository        the user repository
+     */
     public TransactionService(TransactionRepository transactionRepository, UserRepository userRepository) {
         this.transactionRepository = transactionRepository;
         this.userRepository = userRepository;
     }
 
     /**
-     * Create method to do the actual transaction contaning the business logic to add deposit and withdraw and return response according
+     * Create method to do the actual transaction containing the business logic to add deposit and withdraw and return response according.
```
Comment (Ketan-SP18) [inline src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:34]: next time please do changes related to your pr only.

File: src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * The type Transaction service test.
```
Comment (Ketan-SP18) [inline src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java:16]: write your own java doc

File: src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+        when(transactionRepository.findById(transactionID)).thenReturn(Optional.ofNullable(user));
+
+        String response = transactionService.deleteTransaction(transactionID);
+        assertEquals("Your TransactionID " + transactionID + " delete successfully.", response);
+
+    }
+
+    /**
+     * If transaction id does not exist the delete transaction is unsuccessful.
```
Comment (Ketan-SP18) [inline src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java:48]: correct your sentence

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 10]
```diff
@@ -19,13 +20,18 @@ public class TransactionService {
 
     private final UserRepository userRepository;
 
+    /**
+     * Instantiates a new Transaction service.
+     * @param transactionRepository the transaction repository
+     * @param userRepository        the user repository
```
Comment (Ketan-SP18) [inline src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:26]: format it

File: src/main/java/com/passbook/sparkeighteen/controller/TransactionController.java [hunk 11]
```diff
@@ -22,12 +23,17 @@
 public class TransactionController {
     private final TransactionService transactionService;
 
+    /**
+     * Instantiates a new Transaction controller.
+     *
```
[the remark this answers is not in the document]
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/controller/TransactionController.java]: Removed.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java (hunk 7 above)
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:34]: there was a spelling mistake so I changed, next time won't change anything unrelated to my PR.

File: src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java (hunk 8 above)
Comment (Aashishverma06) [reply src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java]: yes it was auto generated, made the required changes.

File: src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java (hunk 9 above)
Comment (Aashishverma06) [reply src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java]: Yes changed.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java (hunk 10 above)
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java]: Done.

**Review by Ketan-SP18 — APPROVED** [review]
Approved

**Review by Hrishikesh-SP18 — APPROVED** [review]

**Comment by kunalcodess** [comment]
Resolve conflicts and get this merged

**Comment by Aashishverma06** [comment]
Resolved all conflict.

**Pull request merged**